addons/mail/static/src/core/web/messaging_menu.js

```diff
@@ -66,6 +72,79 @@ export class MessagingMenu extends Component {
         }
     }
 
+    navigate(direction) {
+        if (this.notificationItems.length === 0) {
+            return;
+        }
+        const activeOptionId = this.state.activeIndex !== null ? this.state.activeIndex : 0;
+        let targetId = undefined;
+        switch (direction) {
+            case "first":
+                targetId = 0;
+                break;
+            case "last":
+                targetId = this.notificationItems.length - 1;
+                break;
+            case "previous":
+                targetId = activeOptionId - 1;
+                if (targetId < 0) {
+                    this.navigate("last");
+                    return;
+                }
+                break;
+            case "next":
+                targetId = activeOptionId + 1;
+                if (targetId > this.notificationItems.length - 1) {
+                    this.navigate("first");
+                    return;
+                }
+                break;
+            default:
+                return;
+        }
+        this.state.activeIndex = targetId;
+        this.notificationItems[targetId].scrollIntoView({ block: "nearest" });
+    }
+
+    onKeydown(ev) {
+        if (!this.dropdown.isOpen) {
+            return;
+        }
+        this.isKeyboardNavigating = true;
+        const hotkey = getActiveHotkey(ev);
+        switch (hotkey) {
+            case "enter":
+                if (this.state.activeIndex === null) {
+                    this.dropdown.close();
+                    return;
+                }
+                this.notificationItems[this.state.activeIndex].click();
```

Comment:
missing stoppropagation here, when having focus of other stuff, for example: a focus on an application, and click enter, you go into the app and open the chatwindow at the same time.